src/util/UniqueCleanup.h

```diff
@@ -49,7 +49,12 @@ class UniqueCleanup {
 
   /// Runs the cleanup call preemptively by moving out of this instance and
   /// immediately destructing the new instance
-  void invokeManuallyAndCancel() { auto self = std::move(*this); }
+  void invokeManuallyAndCancel() {
+    if (active_) {
+      std::invoke(function_, std::move(value_));
```

Comment:
For the same reason as the previous comment, I wouldn't `move` here, but only pass in the reference I think this makes itu usable in more scenarios.